dashboard/app/models/script.rb

```diff
@@ -938,7 +969,13 @@ def self.add_script(options, raw_lesson_groups, raw_lessons, new_suffix: nil, ed
     non_lockable_count = 0
 
     unless raw_script_levels.empty?
-      if raw_lesson_groups.empty?
+      # We want a script to either have all of its lessons have lesson groups
+      # or none of the lessons have lesson groups. We know we have hit this case if
+      # there are more than one lesson group for a script but the lesson group key
+      # for the first lesson is blank
+      if raw_lesson_groups[0][:key].nil? && raw_lesson_groups.length > 1
```

Comment:
It seems like we might want to protect against getting into this state earlier, if we can, since the thing passed into `add_script` is now `raw_lesson_groups`, not lessons themselves. So it seems like at this point every lesson should already belong to a lesson group.

More generally I think it's a good goal to minimize the parts of the code that need to know or care about the "default lesson group" logic. I think it'd be reasonable to put that in the Script DSL side of things, since you could consider not specifying a lesson group to just be a shorthand for specifying an unnamed lesson group.

Reply:
I think some of the later work will hopefully address this concern but if it does not let me know.